### PR TITLE
Fix: order recurring options before updating

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This fix ensures the recurring period options are in the appropriate order (time-frame ascending) - regardless of the time/order in which you toggle them.

## Affects

The recurring billing period options

## Visuals

Screenshot 2023-06-14 at 1 36 16 PM

## Testing Instructions

Toggle the recurring billing periods in different orders and make sure they are in ascending order (in terms of time frame)

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204414138863604

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@pauloiankoski this is ready for review.

File: src/FormBuilder/resources/js/form-builder/src/blocks/fields/amount/inspector/index.tsx [hunk 1]
```diff
@@ -34,18 +47,32 @@ const Inspector = ({attributes, setAttributes}) => {
...
+        (value) => {
+            const options = getOrderedRecurringBillingPeriodOptions(
+                recurringBillingPeriodOptions.filter((option) => option !== value)
+            );
+
+            if (recurringBillingPeriodOptions.length > 1) {
+                setAttributes({
+                    recurringBillingPeriodOptions: options,
+                });
+            }
```
Comment (pauloiankoski) [inline src/FormBuilder/resources/js/form-builder/src/blocks/fields/amount/inspector/index.tsx:72]:
```suggestion
            const options = recurringBillingPeriodOptions.filter((option) => option !== value);

            if (recurringBillingPeriodOptions.length > 1) {
                options.sort(compareBillingPeriods);

                setAttributes({
                    recurringBillingPeriodOptions: options,
                });
            }
```

**Review by pauloiankoski — COMMENTED** [review]
Hey @jonwaldstein, I'm sharing some ideas to reach the same goal while keeping things clear and simple. Could you let me know your opinion?

**Comment by jonwaldstein** [comment]
> Hey @jonwaldstein, I'm sharing some ideas to reach the same goal while keeping things clear and simple. Could you let me know your opinion?

Thanks for the suggestions, I was skeptical on using sort but you proved me wrong - worked like a charm!

**Review by pauloiankoski — APPROVED** [review]
👍🏻

**Pull request merged**